### PR TITLE
[runtime] Add configuration for the default sgen gc mode

### PR DESCRIPTION
**Pull request by alexanderkyte**

### CODE REVIEW

File: autogen.sh [hunk 1]
```diff
@@ -163,6 +163,12 @@ fi
 
 conf_flags="--enable-maintainer-mode --enable-compile-warnings" #--enable-iso-c
 
+echo "MONO_CONC_GC is $MONO_CONC_GC"
```
Comment (kumpera) [inline autogen.sh]: How about a MONO_EXTRA_CONFIGURE_FLAGS to make this not a one time off?

File: configure.ac [hunk 2]
```diff
@@ -3262,15 +3262,21 @@ dnl
...
-   SGEN_DEFINES="-DHAVE_SGEN_GC"
    if test "x$gc_msg" = "x"; then
       gc_msg="sgen"
    else
       gc_msg="sgen and $gc_msg"
    fi
+
+   if test x$with_sgen_default_concurrent = xno; then
+       SGEN_DEFINES="-DHAVE_SGEN_GC"
+   else
```
Comment (kumpera) [inline configure.ac]: We should use AC_DEFINE for this define as it doesn't interfere with boehm.

Additionally, it should be namespaced.

File: autogen.sh (hunk 1 above)
Comment (vargaz) [inline autogen.sh]: Why is this line needed ?

File: autogen.sh [hunk 3]
```diff
@@ -163,6 +163,12 @@ fi
...
+
+if test x$MONO_CONC_GC != x; then
+conf_flags="$conf_flags --with-sgen-default-concurrent"
+fi
+
```
Comment (vargaz) [inline autogen.sh]: Why is this needed ? The user can just pass the same flag to autogen.sh.

Comment (alexanderkyte) [inline autogen.sh]: We're using an environment variable to make it easy to inject config flags into the build.

Comment (vargaz) [inline autogen.sh]: I don't think we should add an env variable for one of the mirriad configure options we have.

**Comment by kumpera** [comment]
This is why I think we should have an env var that can include any extra flags.

We need this so we can pass extra flags on wrench.

**Comment by vargaz** [comment]
That should be added to bockbuild. Mono's configure already takes its options on its own command line.
